firmware/projects/TMS/main.cc

```diff
@@ -113,26 +132,33 @@ TempSensor temp_sensors[] = {
 const int kSensorCount = 6;
 TempSensorManager<kSensorCount> ts_manager{temp_sensors};
 
+BmsBroadcaster bms_broadcaster(veh_can_bus, kSensorCount);
+
 /***************************************************************
     Program Logic
 ***************************************************************/
 
 void Update() {
     static float temperature_buffer[kSensorCount];
+    static uint8_t low_thermistor_idx;
+    static uint8_t high_thermistor_idx;
 
+    veh_can_bus.Update();
     ts_manager.Update();
     ts_manager.GetTemperatures(temperature_buffer);
 
-    float temp_min =
-        shared::util::GetMinimum<float, kSensorCount>(temperature_buffer, NULL);
+    float temp_min = shared::util::GetMinimum<float, kSensorCount>(
+        temperature_buffer, &low_thermistor_idx);
 
-    float temp_max =
-        shared::util::GetMaximum<float, kSensorCount>(temperature_buffer, NULL);
+    float temp_max = shared::util::GetMaximum<float, kSensorCount>(
+        temperature_buffer, &high_thermistor_idx);
 
     float temp_avg =
         shared::util::GetAverage<float, kSensorCount>(temperature_buffer);
 
-    /// TODO: Pack & send CAN message
+    bms_broadcaster.SendBmsBroadcast(
+        high_thermistor_idx, static_cast<int8_t>(temp_max), low_thermistor_idx,
+        static_cast<int8_t>(temp_min), static_cast<int8_t>(temp_avg));
```

Comment:
SInce this is an argument-heavy method, add a trailing comma after the last argument, then clangd will nicely format the  arguments on individual rows. This will help, especially since clangd will then label the arguments.

Reply:
I remembered you had told me how to format it but couldnt remember what it was I had to add. Thanks